rope_.d

```diff
@@ -74,19 +74,19 @@ template Rope(T,S=void)if(is(S==void) || is(typeof((S a,S b)=>a.combine(b)))){
 		static if(segtree){
 			@property value(){ return isArray()?tree?tree.value:S.init:rope.value; }
 			private this(T[] array, SliceSegTree* tree){
-				if(array==[]) return;
 				this.array = array;
+				if(array==[]) return;
 				this.tree = tree;
 			}
 		}
 
 		private this(RopeImpl* rope){ this.rope = rope; }
 		invariant(){ assert(cast(void*)rope is array.ptr); }
-		private union{
+		/*private*/ union{
```

Comment:
Sorry, I was not clear enough. This is a hijacking situation. It's not supposed to be accessible from semantic. The .array expressions in semantic are calls to std.array.array.